### DISCUSSION TITLE
Updated "ooba" BLAS = 0 again? I had already setup GGML with GPU support on Windows 10. Not working now.

### DISCUSSION THREAD

**Discussion by viperwasp** [oobabooga/text-generation-webui#2774] [General]
I did not change drivers, Visual Studios, Nvidia Toolkit or anything. The only thing I did is run the "Ooba" updater. I think I should maybe the the GPU activation steps again. But if anyone has other ideas or if this is a bug let me know.

**Comment by viperwasp** [comment]
I found the "fix" 
Load correct Conda environment. 
First run
pip uninstall -y llama-cpp-python

And then run each of the following one line at a time 
set FORCE_CMAKE=1
set "CMAKE_ARGS=-DLLAMA_CUBLAS=on"
set "CUDAFLAGS=-arch=all -lcublas"
python -m pip install git+https://github.com/abetlen/llama-cpp-python

Ran "update_windows" again to see if it brings back the problem. 
It did not. So all seems well. 

By the way I don't know why but the follow code below did not work. I got a weird error I don't understand. 
pip uninstall -y llama-cpp-python
set LLAMA_CUBLAS=1
set CMAKE_ARGS="-DLLAMA_CUBLAS=on"
set FORCE_CMAKE=1
pip install llama-cpp-python --no-cache-dir

But I ran the code I gave for the fix and it worked. If anyone understands this feel free to chime in. Otherwise this might help others if BLAS = 0 after update.

**Reply by ossirytk** [reply]
I had the same problem yesterday. Adding the `set "CUDAFLAGS=-arch=all -lcublas"` seems to be a new thing now. Oddly this is only required for ooba. If you build the llama-cpp-python alone, this is not needed. I'm not sure what changed to require this.

Ps. I don't use `python -m pip install git+https://github.com/abetlen/llama-cpp-python`. This might be good form to ensure that the version is correct, but usually it's the latest release, so `pip install llama-cpp-python` is ok. I don't use the python version either since this runs in a virtual env, so this should always be the correct python anyways.

**Comment by jparmstr** [comment]
I've had a similar problem. llama-cpp-python can no longer be compiled with CUBLAS support after version 0.1.62 using the [instructions ](https://github.com/oobabooga/text-generation-webui/blob/main/docs/llama.cpp-models.md) that previously worked. This isn't isolated to a specific version of llama-cpp-python, it's affected every version newer than 0.1.62.

It's normal, just inconvenient, that running the update_windows script erases CUBLAS support and you have to manually rebuild it.

These were the minimal instructions that worked for me to successfully build as of version 0.1.65:
```
pip uninstall -y llama-cpp-python
set CMAKE_ARGS="-DLLAMA_CUBLAS=on"
set FORCE_CMAKE=1
python -m pip install git+https://github.com/abetlen/llama-cpp-python
```
